core/chains/evm/txmgr/eth_confirmer.go

```diff
@@ -1102,23 +1109,27 @@ func (ec *EthConfirmer[ADDR, TX_HASH, BLOCK_HASH]) markForRebroadcast(etx EthTx[
 // This operates completely orthogonal to the normal EthConfirmer and can result in untracked attempts!
 // Only for emergency usage.
 // This is in case of some unforeseen scenario where the node is refusing to release the lock. KISS.
-func (ec *EthConfirmer[ADDR, TX_HASH, BLOCK_HASH]) ForceRebroadcast(beginningNonce int64, endingNonce int64, gasPriceWei uint64, address ADDR, overrideGasLimit uint32) error {
-	ec.lggr.Infof("ForceRebroadcast: will rebroadcast transactions for all nonces between %v and %v", beginningNonce, endingNonce)
+func (ec *EthConfirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) ForceRebroadcast(seqs []SEQ, gasPriceWei uint64, address ADDR, overrideGasLimit uint32) error {
+	if len(seqs) == 0 {
+		ec.lggr.Infof("ForceRebroadcast: No sequences provided. Skipping")
+		return nil
+	}
+	ec.lggr.Infof("ForceRebroadcast: will rebroadcast transactions for all sequences between %v and %v", seqs[0], seqs[len(seqs)-1])
 
-	for n := beginningNonce; n <= endingNonce; n++ {
+	for _, seq := range seqs {
```

Comment:
just curious, why are we changing logic? i thought this PR was only for type changes

Reply:
This for loop iterates and increments nonces. NextNonce = previousNonce + 1
This assumes nonces are integers, and next one is +1.
Once you generalize nonce into a generic Sequence type, we can no longer trivially get the next Sequence from a previous Sequence.
Previously this function received startingNonce and endingNonce values, and created all nonces. But now it can't.
So the caller has to pass in all Sequence objects in an array.

Another way could be, for the Sequence type to expose another function, saying getNextSequence(). But that's too complex, and may require even more inputs. For example, given a Bitcoin UTXO, you cannot trivially compute the next UTXO without first consuming current one with a Tx.


So overall, this above workaround side-steps this problem.

Reply:
makes sense to accommodate generic types 👍🏼

Reply:
Eventually, we may be able to use range over custom iterator types: https://github.com/golang/go/discussions/56413